diff -up minio-go-6.0.2/api-put-object.go.fix minio-go-6.0.2/api-put-object.go
--- minio-go-6.0.2/api-put-object.go.fix	2018-06-15 01:22:29.559141940 -0400
+++ minio-go-6.0.2/api-put-object.go	2018-06-15 01:23:47.353815108 -0400
@@ -28,7 +28,7 @@ import (
 
 	"github.com/minio/minio-go/pkg/encrypt"
 	"github.com/minio/minio-go/pkg/s3utils"
-	"golang.org/x/net/http/httpguts"
+	"golang.org/x/net/lex/httplex"
 )
 
 // PutObjectOptions represents options specified by user for PutObject call
@@ -101,10 +101,10 @@ func (opts PutObjectOptions) Header() (h
 // validate() checks if the UserMetadata map has standard headers or and raises an error if so.
 func (opts PutObjectOptions) validate() (err error) {
 	for k, v := range opts.UserMetadata {
-		if !httpguts.ValidHeaderFieldName(k) || isStandardHeader(k) || isSSEHeader(k) || isStorageClassHeader(k) {
+		if !httplex.ValidHeaderFieldName(k) || isStandardHeader(k) || isSSEHeader(k) || isStorageClassHeader(k) {
 			return ErrInvalidArgument(k + " unsupported user defined metadata name")
 		}
-		if !httpguts.ValidHeaderFieldValue(v) {
+		if !httplex.ValidHeaderFieldValue(v) {
 			return ErrInvalidArgument(v + " unsupported user defined metadata value")
 		}
 	}
